### PR TITLE
Fix controller specs

### PR DESCRIPTION
**Pull request by KirillKayumov**

### CODE REVIEW

File: frontend/spec/controllers/spree/checkout_controller_spec.rb [hunk 1]
```diff
@@ -170,11 +170,6 @@ def spree_post_address
           expect(response).to redirect_to spree.order_path(order)
         end
 
-        it 'populates the flash message' do
-          post :update, params: { state: 'confirm' }
-          expect(flash.notice).to eq(Spree.t(:order_processed_successfully))
```
Comment (KirillKayumov) [inline frontend/spec/controllers/spree/checkout_controller_spec.rb]: This flash message is not being set in the controller anymore

File: frontend/spec/controllers/spree/home_controller_spec.rb [hunk 2]
```diff
@@ -1,14 +1,6 @@
 require 'spec_helper'
 
 describe Spree::HomeController, type: :controller do
-  it 'provides current user to the searcher class' do
-    user = mock_model(Spree.user_class, last_incomplete_spree_order: nil, spree_api_key: 'fake')
-    allow(controller).to receive_messages try_spree_current_user: user
-    expect_any_instance_of(Spree::Config.searcher_class).to receive(:current_user=).with(user)
```
Comment (KirillKayumov) [inline frontend/spec/controllers/spree/home_controller_spec.rb]: We don't initialize search in the `HomeController` anymore, so these specs can be deleted

File: frontend/spec/controllers/spree/checkout_controller_spec.rb (hunk 1 above)
Comment (damianlegawiec) [reply frontend/spec/controllers/spree/checkout_controller_spec.rb]: @KirillKayumov setting this flash message is essential for conversion tracking - https://github.com/spree/spree/blob/230f6d8e780c5b4ba4173956e83026650e01b4db/frontend/app/helpers/spree/orders_helper.rb#L4

We cannot remove it, we need to preserve it but not display it on Order Confirmation

see https://github.com/spree-contrib/spree_analytics_trackers/search?q=order_just_completed%3F%28%40order%29&unscoped_q=order_just_completed%3F%28%40order%29

Comment (KirillKayumov) [reply frontend/spec/controllers/spree/checkout_controller_spec.rb]: @damianlegawiec thanks but we still have this flag in `flash` https://github.com/spark-solutions/spree/blob/f3fa6bd35364754d9deb765400db823ea980d578/frontend/app/controllers/spree/checkout_controller.rb#L42
so it should work well

**Pull request merged**